src/pages/signin/SignInPage.js

```diff
@@ -35,153 +32,149 @@ const propTypes = {
-        // - AND the login is not validated
-        const showUnlinkLoginForm =
-            Boolean(this.props.credentials.login && this.props.account.primaryLogin) && this.props.account.primaryLogin !== this.props.credentials.login && !this.props.account.validated;
-
-        // Show the old password form if
-        // - A login has been entered
-        // - AND an account exists and is validated for this login
-        // - AND a password hasn't been entered yet
-        // - AND haven't forgotten password
-        // - AND the login isn't an unvalidated secondary login
-        // - AND the user is NOT on the passwordless beta
-        const showPasswordForm =
-            Boolean(this.props.credentials.login) &&
-            this.props.account.validated &&
-            !this.props.credentials.password &&
-            !this.props.account.forgotPassword &&
-            !showUnlinkLoginForm &&
-            !Permissions.canUsePasswordlessLogins(this.props.betas);
-
-        // Show the new magic code / validate code form if
-        // - A login has been entered or a validateCode has been cached from sign in link
-        // - AND the login isn't an unvalidated secondary login
-        // - AND the user is on the 'passwordless' beta
-        const showValidateCodeForm =
-            Boolean(this.props.credentials.login || this.props.credentials.validateCode) && !showUnlinkLoginForm && Permissions.canUsePasswordlessLogins(this.props.betas);
-
-        // Show the resend validation link form if
-        // - A login has been entered
-        // - AND is not validated or password is forgotten
-        // - AND the login isn't an unvalidated secondary login
-        // - AND user is not on 'passwordless' beta
-        const showResendValidationForm =
-            Boolean(this.props.credentials.login) &&
-            (!this.props.account.validated || this.props.account.forgotPassword) &&
-            !showUnlinkLoginForm &&
-            !Permissions.canUsePasswordlessLogins(this.props.betas);
-
-        let welcomeHeader = '';
-        let welcomeText = '';
-        if (showValidateCodeForm) {
-            if (this.props.account.requiresTwoFactorAuth) {
-                // We will only know this after a user signs in successfully, without their 2FA code
-                welcomeHeader = this.props.isSmallScreenWidth ? '' : this.props.translate('welcomeText.welcomeBack');
-                welcomeText = this.props.translate('validateCodeForm.enterAuthenticatorCode');
+    useEffect(() => Performance.measureTTI(), []);
+    useEffect(() => {
+        App.setLocale(Localize.getDevicePreferredLocale());
+    }, []);
+
+    const {
+        shouldShowLoginForm,
+        shouldShowUnlinkLoginForm,
+        shouldShowPasswordForm,
+        shouldShowValidateCodeForm,
+        shouldShowResendValidationForm,
+        shouldShowWelcomeHeader,
+        shouldShowWelcomeText,
+    } = getRenderOptions({
+        hasLogin: Boolean(credentials.login),
```

Comment:
but here we dont have it, I have tried changing this to `Boolean(credentials.login && account.primaryLogin)`, but it did not fix the issue on its own and because its a deploy blocker and there is couple more I have went with revert.